src/strategy/periodic.rs

```diff
@@ -22,7 +23,8 @@ impl StrategyPeriodic {
     pub const LABEL: &'static str = "periodic";
 
     /// Build a new periodic strategy.
-    pub fn new(cfg: inputs::UpdateInput) -> Fallible<Self> {
+    #[context("failed to build new periodic strategy")]
```

Comment:
I'd suggest something like: `parsing periodic updates strategy`.

Reply:
One data point here: the bootupd/rpm-ostree wording in error contexts is semi-predicated on the idea that we'll flatten the error message to a single line.  IOW an error would look like:

```
error: Updating: Post-transaction cleanup: Removing /var/lib/somedata: Input/output error
```

The default anyhow formatter though really wants to preserve the full cause chain - but the way it prints looks weird unless each string stands on its own:

```
error: Failed to update
Caused by:
  0: Failed post-transaction cleanup
  1: Couldn't remove /var/lib/somedata
  2: Input/output error
```

In this model the formatting hence IMO looks weird unless *every* context message includes the string "failed" or otherwise makes clear it's an error.

I think this can be kind of mitigated by ensuring that the final error has `error: ` as a prefix.  In this case if it was just "parsing periodic updates strategy" then you'd see:

`error: parsing periodic updates strategy`
which isn't bad at all, but IMO reads better as `error: Failed to parse periodic updates strategy`.

Reply:
Admittedly, I hadn't thought about each string standing on its own vs flattening the error message to a single line, I just mimicked the style in the `anyhow` docs. But IIUC, Zincati logs the errors to the journal by chaining them like this https://github.com/coreos/zincati/blob/3a88c05fae179cc467f6968362c0f00077261b16/src/main.rs#L65, which I think is closer to how the defauly `anyhow` formatter prints errors right? It looks something like
```
[ERROR] critical error: failed to assemble configuration settings
[ERROR]  -> failed to validate update-strategy configuration
[ERROR]  -> No such file or directory (os error 2)
```
To me, actually, both styles seem to make sense to me, so probably the most important thing is to be consistent. Based on what I've seen, it looks like in majority of other places we are doing `.context("failed to run 'rpm-ostree' binary")?;`.

Reply:
Uhm yeah, we should probably do a project-wide sweep. Right now Zincati critical errors are logged like this:
```
[INFO ] starting update agent (zincati 0.0.19)
[ERROR] critical error: failed to assemble configuration settings
[ERROR]  -> failed to validate agent identity configuration
[ERROR]  -> unexpected negative rollout wariness: -0.002
```

But transient errors (e.g. from remote HTTP requests or from OS calls) are instead logged in the single-line format, like this:
```
[ERROR] failed to check Cincinnati for updates: client-side error: error sending request for url (https://updates.coreos.fedoraproject.org/...): error trying to connect: Connection reset by peer (os error 104)
```

I guess that contrary to my initial comment, sticking to the format of `failed to perform xyz` is the best way to get a uniform result.